### ISSUE TITLE
poor quality traceback / useful stack frames not present when exceptions raised in sync dependencies

### ISSUE THREAD

**Issue by cjw296** [fastapi/fastapi#13067]
### Privileged issue

- [X] I'm ~@tiangolo or he asked me directly to create an issue here~ a liar, but the discussion template was crazy onerous, and I'm confident I can write a decent, succinct issue description that's worth reading here ;-)

### Issue Content

If I apply this diff to the [full-stack-fastapi-template](https://github.com/fastapi/full-stack-fastapi-template/blob/88c83cc06ccab67efa839c2c0994435b727986a3/backend/app/api/deps.py#L21-L23):

```diff
diff --git a/backend/app/api/deps.py b/backend/app/api/deps.py
index c2b83c8..c99cdb2 100644
--- a/backend/app/api/deps.py
+++ b/backend/app/api/deps.py
@@ -19,6 +19,7 @@ reusable_oauth2 = OAuth2PasswordBearer(
 
 
 def get_db() -> Generator[Session, None, None]:
+    raise Exception("error")
     with Session(engine) as session:
         yield session
```

...and issue a GET to `http://127.0.0.1:8000/api/v1/users/`, the traceback shown in the console is really unhelpful:

```
  File "...3.11/lib/python3.11/contextlib.py", line 650, in enter_async_context
    result = await _enter(cm)
             ^^^^^^^^^^^^^^^^
  File "...3.11/lib/python3.11/contextlib.py", line 210, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File ".../full-stack-fastapi-template/backend/.venv/lib/python3.11/site-packages/fastapi/concurrency.py", line 35, in contextmanager_in_threadpool
    raise e
Exception: error

```

There are no frames from the actual site of the exception, and, on the face of it, potentially no way to go back from the exception to the source of the error.

I first noticed this doing test driven development on a new FastAPI project, having not done any FastAPI dev for a couple of years, and was pretty shocked.

What I don't understand is why the exception being re-raised appears to have no traceback of its own?

**Comment by Kludex** [comment]
Did you try the last FastAPI version?

**Comment by cjw296** [comment]
Close enough:

```
$ uv tree | grep -i fast
Resolved 81 packages in 0.92ms
├── fastapi[standard] v0.115.0
│   ├── fastapi-cli[standard] v0.0.5 (extra: standard)
├── sentry-sdk[fastapi] v1.45.1
│   └── fastapi v0.115.0 (extra: fastapi) (*)
```


Dropped a breakpoint, does indeed appear that `__traceback__` is None, but why?

```
> .../python3.11/site-packages/fastapi/concurrency.py(36)contextmanager_in_threadpool()
-> raise e
(Pdb) !e
Exception('error')
(Pdb) !e.__traceback__
(Pdb) !e.__traceback__ is None
True
```

**Comment by Kludex** [comment]
Can you try on the really latest?

**Comment by Kludex** [comment]
For reference, this issue was solved in the last release: https://github.com/fastapi/fastapi/releases/tag/0.115.6

- Specifically on https://github.com/fastapi/fastapi/pull/5823.

I was on my phone, couldn't get the reference.

---

About this:

<img width="772" alt="Screenshot 2024-12-13 at 10 01 20" src="https://github.com/user-attachments/assets/cd2397ec-7e65-449b-9501-62c2ba6ae027" />

It's not about your "issue creation skills", it's about project organization. It helps to have a flow where I can triage before moving to issues.

**Comment by cjw296** [comment]
@Kludex - about the issue/discussions thing... if you want people to use the discussions, please make the template significantly simpler. Having to check 8 check boxes before I can type any useful text is just silly. The folks who you're trying to get to think first will just click through them anyway, for the rest of us it's pretty annoying.

**Comment by Kludex** [comment]
I don't really have the power or motivation to do that. **I don't maintain FastAPI.**

If you want, you can create a GitHub issue about the "8 checks on the discussion". I'll not close it, so @tiangolo can see later.

**Comment by cjw296** [comment]
Anyway, confirmed fixed in 0.115.6
